include/boost/url/experimental/detail/vformat.hpp

```diff
@@ -0,0 +1,119 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+#ifndef BOOST_URL_EXPERIMENTAL_DETAIL_FORMAT_HPP
+#define BOOST_URL_EXPERIMENTAL_DETAIL_FORMAT_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/string_view.hpp>
+#include <boost/url/url.hpp>
+#include <boost/url/experimental/detail/format_args.hpp>
+#include <functional>
```

Comment:
is there way to avoid including this header?

Reply:
I think it's worth including it because we would be replicating most of the functionality if we didn't.

Reply:
Do users end up seeing it? What do we need from it?

Reply:
When implementing our mini-fmtlib, the format `args` are type-erased with a `handle` type that can format the string for whatever the original type was. Then I used `std::function` for that because a custom type would be 90% the same as `std::function`. 

However, because `function` is being used to type erase the formatter, they are always the same types. That means we can get rid of any extra compile time spent on instantiations with `extern template` in the `ipp` file.

Reply:
I'll think about it but I'm not sure how I can hide that yet. Unless I reimplement my own version of the `function<...>` overloads I need.

Reply:
What's wrong with `extern template` ?

Reply:
This can be done with a plain function pointer instead of `std::function` but also it could use an `initializer_list` to avoid the need to allocate memory. Follow me for more C++ Tips.